ark/utils/data_utils.py

```diff
@@ -4,6 +4,74 @@
 import numpy as np
 import xarray as xr
 
+from ark import settings
+from ark.utils.misc_utils import verify_in_list
+
+
+def relabel_images(fovs, all_data, label_maps, cell_label_col='fovs',
+                   fov_col=settings.FOV_ID):
+    """ Translates cell-ID labeled images according to the clustering assignment.
+
+    Takes a list of fovs, and relabels each image (array) according to the assignment
+    of cell IDs to cluster label.
+
+    Args:
+        fovs (list):
+            List of fovs to relabel.
+        all_data (pandas.DataFrame):
+            data including fovs, cell labels, and cell expression matrix for all markers.
+        label_maps (xr.DataArray):
+            xarray of label maps for multiple fovs
+        cell_label_col (str):
+            column with the fov names in label_maps.
+        fov_col (str):
+            column with the fovs names in all_data.
+    Returns:
+        xr.DataArray:
+            The relabeled images (dims: ["fovs", "rows", "cols"]).
+    """
+
+    # check if included fovs found in fov_col
+    verify_in_list(fov_names=fovs, unique_fovs=all_data[fov_col].unique())
+    verify_in_list(fov_names=fovs, unique_fovs=label_maps[cell_label_col])
+
+    img_data = []
+    for fov in fovs:
+        df = all_data[all_data[fov_col] == fov]
+        labels_dict = dict(zip(df.label, df.cluster_labels))
```

Comment:
df.label and df.cluster_labels should be changed to use the relevant global variables and pulled out as arguments to the function; these can change from one person's dataset to another

Reply:
changed